### ISSUE TITLE
Enable deletion of language tree nodes

### ISSUE THREAD

**Issue by timoludwig** [Integreat/integreat-cms#505]
### Motivation

At the moment, language tree nodes can only be deleted via the Django Admin backend

### Proposed Solution

Add the possibility to delete language tree nodes via the language tree or the language tree node form.

### Additional Context

Also keep in mind what to do if content in this language exists.
Since the content is only related to the language objects in the database (not the language tree nodes), there is no inbuilt deletion protection here.

**Comment by melegiul** [comment]
What's about the descendants of a language tree node? Should they be deleted too or is deletion blocked as long as there exist children of a language tree node (like the page deletion)?

**Comment by timoludwig** [comment]
The deletion should be blocked as long as there are descendants.

**Comment by melegiul** [comment]
I tried to stick close close to the projects conventions for confirmation popups, but I encountered some redundant templates for the popup dialogs. For each template folder like pages, pois, events, etc. there is a subdirectory for a confirmation dialog with three different templates. These templates are quiet the same, they only differ in single words like 'Are you sure you want to delete/archive/restore?'. So since I need myself a confirmation dialog for deleting the language nodes, I suggest a generic central template called 'confirmation-dialog'. This template is empty by default and only gets populated at runtime with the appropriate text according to current task (delete/archive/restore). What you think about this?
